### ISSUE TITLE
Polish the plain text pasting

### ISSUE THREAD

**Issue by jodator** [ckeditor/ckeditor5#7799]
I'm creating a ticket to follow-up on @Reinmar from the #7778 PR.

> 1.  Firefox pastes `foo` as both `text/html` and `text/plain` - a simple string comparison would be enough to set `isPlainText`

Good idea, we could consider that. That'd be a bit more complex as the HTML could look like this: `'foo &nbsp;bar&gt;bom'` (`foo  bar<bom`) but with some simple regexp replace it's doable.

As for the other things that @jodator wrote, I think that we need to be predictable:

*   If you pressed ctrl+shift+v it's very likely that you want to inherit selection attributes. But then @pomek's solution won't handle multi-line text as in the model it'd contain soft breaks or paragraphs. Do we want to handle multi-line text the same way as a single line? If yes, we need an additional check for that.
*   If you pressed ctrl+v but you had very simple content in the clipboard, how likely is it that you want it to inherit selection styles? I'd say, quite likely, so I found @pomek's idea really interesting. However, I'm slightly worried about the randomness of this solution. I copied some formatted text from paste from office or other application and that formatting is not supported by the editor – that will make this text to inherit the selection styles. But if a supported style was included in that text, the style will not be inherited. It may feel "random". But that's where I count on your intuition and experience, @Mgsy. Can we live with that?

_Originally posted by @Reinmar in https://github.com/ckeditor/ckeditor5/pull/7778#issuecomment-669833922_

---

Actions needed to close:

- [ ] 1. check multi-line, plain text <kbd>shift+ctrl+v</kbd> scenario behavior - what is in the clipboard? Add tests. // cc @pomek
- [ ] 2. upon that - yes/no for inheriting text attributes on pasting multi-line text using <kbd>shift+ctrl+v</kbd> // waiting for :point_up:
- [ ] 3. decide on "random" scenario of pasting unsupported formatting resulting in plain text-like paste // cc @Mgsy

**Comment by Reinmar** [comment]
Notes from a sync:

* We agreed that for us it'd be intuitive to treat multi-line plain text the same way that we treat HTML content right now – so, to not inherit the styles from the current selection position.
* However, we should check how cdm+shift+v works in other editors, as it's not clear for us whether "paste and match style" means that the content that was pasted should inherit the current selection position styles or not. Assuming that it should, we should add support for that.

**Comment by mlewand** [comment]
Few notes on how other editors handle it.

tl;dr: It makes a perfect sense to keep the styles when pasting with explicit _paste as plain text_ command. All editors do that.

Also multiline case have a common handling in mentioned editors.

## Paste as plain text (ctrl/cmd+shift+v) behavior

### gdocs

| FF | Chrome
---------|----------|---------
single line | inherit | inherit
multiline | inherit | inherit

### Word

This is performed by using Word's paste as plain text option (right click, `Paste Options:` > `Keep Text Only (T)`)

| Word
---------|----------
single line | inherit
multiline | inherit

### CKE4

| FF | Chrome
---------|----------|---------
single line | inherit | inherit
multiline | inherit | inherit


## regular paste

### gdocs

Note it gives different results depending where your "plain" text comes from. I copied non-formatted text in CKE4 instance on Chrome.

| FF | Chrome
---------|----------|---------
single line | inherit | inherit
multiline | inherit | inherit

### Word

| Word
---------|----------
single line | breaks
multiline | breaks

### CKE4

| FF | Chrome
---------|----------|---------
single line | breaks | breaks
multiline | breaks | breaks

**Comment by mlewand** [comment]
btw on a side note, ideally we should expose this feature with some other mans than <kbd>ctrl/cmd</kbd> + <kbd>shift</kbd> + <kbd>v</kbd> keystroke as this makes the feature inaccessible for:

*   mobile users
*   users that do not use keyboard
